### PR TITLE
[5.8] Make sure changing a database field to binary does not include collation

### PR DESCRIPTION
**Pull request by JSRM99**
This PR will fix an issue very close to https://github.com/laravel/framework/issues/25735#issuecomment-423773584. It will fix pretty much the same error but then when changing a text field in to a binary.

### CODE REVIEW

File: src/Illuminate/Database/Schema/Grammars/ChangeColumn.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ protected static function getDoctrineColumnChangeOptions(Fluent $fluent)
             $options['length'] = static::calculateDoctrineTextLength($fluent['type']);
         }
 
-        if ($fluent['type'] === 'json') {
+        if ($fluent['type'] === 'json' || $fluent['type'] === 'binary') {
```
Comment (staudenmeir) [inline src/Illuminate/Database/Schema/Grammars/ChangeColumn.php]: Can you please use `in_array()`?

Comment (JSRM99) [reply src/Illuminate/Database/Schema/Grammars/ChangeColumn.php]: I changed it to use `in_array()`.

**Review by staudenmeir — APPROVED** [review]

File: src/Illuminate/Database/Schema/Grammars/ChangeColumn.php (hunk 1 above)
Comment (ebrigham1) [reply src/Illuminate/Database/Schema/Grammars/ChangeColumn.php]: Should `in_array()` be used in strict mode (pass true for third parameter) or was the type checking not needed in the first place?

Comment (staudenmeir) [reply src/Illuminate/Database/Schema/Grammars/ChangeColumn.php]: I don't see a case where the strict mode would be necessary, we are always comparing strings.

**Pull request merged**